src/CompHash.cc

```diff
@@ -1,24 +1,24 @@
 // See the file "COPYING" in the main distribution directory for copyright.
 
 #include "zeek-config.h"
-#include "zeek/CompHash.h"
 
 #include <cstring>
-#include <vector>
 #include <map>
+#include <vector>
 
-#include "zeek/ZeekString.h"
+#include "zeek/CompHash.h"
```

Comment:
I'd like to figure out why it's moving the local header for a .cc file down into the main block of headers. It's likely something to do with the regexes in the `IncludeCategories` section of the configuration though, and not something with clang-format itself. Changes like this have the chance to break the build, as seen in the last commit on the branch.

Reply:
This is because of the rules in `IncludeCategories`. It's getting caught by this rule:

```
  - Regex: '^"zeek/'
    Priority: 4
```

By default clang-format will automatically set the main header associated with a `cc` file as priority zero and it'll end up at the top of the list. This breaks because of the change to add `zeek/` to the front of all of our includes, including the one for the main header of a source file. There's ways in the clang-format configuration to set additional suffixes for this main header, but none to handle if you've prefixed it with something.